pycloak/middleware.py

```diff
@@ -184,3 +186,56 @@ def allow_default_login(self, request) -> bool:
 
     def get_token_id(self, request, jwt_data) -> str:
         return jwt_data[conf.TOKENID_CLAIM]
+
+    def store_claim_on_user(self, request, user):
+        for claim, attr in conf.CLAIM_TO_USER_MAPPING.items():
+            if not attr.get("field"):
+                raise ImproperlyConfigured(
+                    f"PYCLOAK_CLAIM_TO_USER_MAPPING: {claim} has no field specified"
+                )
+
+            # check for related fields on the user model
+            rel_fields = attr["field"].split(".")
+            field = rel_fields.pop(0)
+            obj = user
+            for rel_field in rel_fields:
+                obj = getattr(obj, rel_field)
+
+            # check for value
+            if value := request.jwt_data.get(claim) is None:
+                logger.warning(f"Claim {claim} not found in jwt")
+                if conf.PYCLOAK_CLAIM_SKIP_MISSING:
+                    continue
+                else:
+                    raise ImproperlyConfigured(f"Claim {claim} not found in jwt")
+
+            # check for callback method
+            if callback_fun := attr.get("callback"):
+                try:
+                    value = callback_fun(value)
+                except Exception as e:
+                    logger.exception(
+                        f"Callback {callback_fun} failed for claim {claim}",
+                        error=str(e),
+                    )
+                    raise ImproperlyConfigured(
+                        f"Callback {callback_fun} failed for claim {claim}"
+                    )
+
+            # explicitly validate value on field
+            # we do this, as we want to save all fields that we can if PYCLOAK_IGNORE_VALIDATION_ERROR is True
+            for obj_field in obj._meta.fields:
+                if field == obj_field.name:
+                    try:
+                        obj_field.get_prep_value(value)
+                        setattr(obj, field, value)
+                    except ValidationError as e:
+                        logger.warning(
+                            f"Validation error for claim {claim} on field {field}",
+                            error=str(e),
+                        )
+                        if not conf.PYCLOAK_CLAIM_IGNORE_VALIDATION_ERROR:
+                            raise e
+
+            # save the object (user or related object)
+            obj.save()
```

Comment:
This is potentially called multiple times. Maybe not ideal?

Reply:
I changed it.